app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java

```diff
@@ -209,6 +220,75 @@ public static void enqueueOnBackgroundPlayer(final Context context,
+    }
+
+    /**
+     * @param context          Context
+     * @param queue            a single video
+     * @param selectOnAppend   play when video is "play nexted"
+     * @param resumePlayback   ?
+     */
+    public static void nextOnVideoPlayer(final Context context,
+                                         final PlayQueue queue,
+                                         final boolean selectOnAppend,
+                                         final boolean resumePlayback) {
+        Toast.makeText(context, R.string.playing_next, Toast.LENGTH_SHORT).show(); //Toast
+        final Intent intent = getPlayerNextIntent(
+                context, MainPlayer.class, queue, selectOnAppend, resumePlayback);
+
+        intent.putExtra(Player.PLAYER_TYPE, MainPlayer.PlayerType.VIDEO.ordinal());
+        ContextCompat.startForegroundService(context, intent);
+    }
+
+    public static void nextOnPopupPlayer(final Context context,
+                                         final PlayQueue queue,
+                                         final boolean resumePlayback) {
+        nextOnPopupPlayer(context, queue, false, resumePlayback);
+    }
+
+    public static void nextOnPopupPlayer(final Context context,
+                                         final PlayQueue queue,
+                                         final boolean selectOnAppend,
+                                         final boolean resumePlayback) {
+        if (!PermissionHelper.isPopupEnabled(context)) {
+            PermissionHelper.showPopupEnablementToast(context);
+            return;
+        }
+        Toast.makeText(context, R.string.playing_next, Toast.LENGTH_SHORT).show();
+        final Intent intent = getPlayerNextIntent(
+                context, MainPlayer.class, queue, selectOnAppend, resumePlayback);
+
+        intent.putExtra(Player.PLAYER_TYPE, MainPlayer.PlayerType.POPUP.ordinal());
+        ContextCompat.startForegroundService(context, intent);
+    }
+
+    public static void nextOnBackgroundPlayer(final Context context,
+                                              final PlayQueue queue,
+                                              final boolean resumePlayback) {
+        nextOnBackgroundPlayer(context, queue, false, resumePlayback);
+    }
+
+    public static void nextOnBackgroundPlayer(final Context context,
+                                              final PlayQueue queue,
+                                              final boolean selectOnAppend,
+                                              final boolean resumePlayback) {
+        Toast.makeText(context, R.string.playing_next, Toast.LENGTH_SHORT).show();
+        final Intent intent = getPlayerNextIntent(
+                context, MainPlayer.class, queue, selectOnAppend, resumePlayback);
+
+        intent.putExtra(Player.PLAYER_TYPE, MainPlayer.PlayerType.AUDIO.ordinal());
+        ContextCompat.startForegroundService(context, intent);
+    }
```

Comment:
Just do not add any of this and add only a `enqueueNextOnPlayer` with this code:
```java
    public static void enqueueNextOnPlayer(final Context context,
                                              final PlayQueue queue,
                                              final boolean selectOnAppend,
                                              final boolean resumePlayback) {
        Toast.makeText(context, R.string.playing_next, Toast.LENGTH_SHORT).show();
        final Intent intent = getPlayerNextIntent(
                context, MainPlayer.class, queue, selectOnAppend, resumePlayback);

        intent.putExtra(Player.PLAYER_TYPE, PlayerHolder.getInstance().getType().ordinal());
        ContextCompat.startForegroundService(context, intent);
    }
```
Assuming it is really needed to pass `PLAYER_TYPE`. Otherwise remove that, too.

---

Edit: just noticed you did this since you took inspiration from `enqueueOn*Player` methods. While you are at it, you could remove them all and create a single `enqueueOnPlayer`, similarly to what I suggested above.

---

Edit 2: I also noticed the `selectOnAppend` parameter is always `false`. Also remove that completely, from `Player`, from every method in `NavigationHelper` and anywhere else.

Reply:
i've removed and merged the redundant `enqueueOn*Player` methods and replaced them with ones without `selectOnAppend` and `RESUME_PLAYBACK`.
`PLAYER_TYPE` is still passed as in your code.

Reply:
enqueueOn*Player methods are needed in other places
will try to resolve this

Reply:
I want to note that `resumePlayback `seems to be always true except for in `VideoDetailFragment.enqueueOnPopupPlayer` where it is false.
Since according to you it is unused and i cant make out any uses;
 I'll replace any method calls if the `enqueueOn*Player` unrelated to if they specify `resumePlayback` as false

Reply:
f7f98518c15ea2a67deab14a73ab574c3807905e removes the remaining calls to the `enqueueOn*Player` methods with the `enqueueOnPlayer` method, independent of the `resumePlayback` parameter.
Thus it can be build again.